### ISSUE TITLE
Flutter 2 and 3 compatibility

### ISSUE THREAD

**Issue by ekuleshov** [marcglasberg/assorted_layout_widgets#20]
Some changes in version 6.0.0 introduced changes specific to Flutter 3 only.

https://github.com/marcglasberg/assorted_layout_widgets/blob/f7176a9dd40f1e0456c7aafafc29c91d33ee07fc/lib/src/delayed.dart#L40

It would be great to support both Flutter 2 and 3 there using approach similar to this

https://github.com/jonataslaw/getx/commit/9d2b87242aa5d5b54aebd4d635d1e16835635c49

More details are at https://docs.flutter.dev/development/tools/sdk/release-notes/release-notes-3.0.0#your-code

**Comment by marcglasberg** [comment]
Hey, thanks for the idea, it's a very interesting one. However, I don't intend to support old Flutter versions. I have 14 packages to maintain, so I must keep it simple to have enough bandwidth. 

If you want to use it with Flutter 2, just fix the version at `5.8.5`, like this:

```
assorted_layout_widgets: 5.8.5
```

**Comment by ekuleshov** [comment]
> If you want to use it with Flutter 2, just fix the version at `5.8.5`

The problem with that is I have to keep multiple versions of Flutter because packages like your aren't backward compatible and not all of them moved to Flutter 3, so some of my projects have to use Flutter 3 and some have to stay with Flutter 2. But all new fixes and new features are only added for Flutter 3.

So, I was really hoping that small fix with "ambiguate" would be a small price maintain backward compatibility with Flutter 2 and support Flutter 3.

**Comment by marcglasberg** [comment]
I understand. But it's better that you ask those other packages to upgrade to Flutter 3.0.

What I do myself is usually waiting a week or 2 before upgrading to a new Flutter version, to give it time for package maintainers to upgrade their packages. In other words, maybe you could keep all your packages in Flutter 2 until all your packages upgrade. Also some packages may never upgrade because they are abandoned. It's a sad reality that we sometimes have to try and find an alternative for those packages.

**Comment by ekuleshov** [comment]
Right. Though I'm still puzzled what are your reservations on making your package backward compatible? The required change seem minor and well isolated. 

PS: btw, your package dart sdk level is still at the version supported by Flutter 2. So, currently it just fail at build time when incompatible Flutter version is used.

**Comment by marcglasberg** [comment]
I don't want to do that, sorry.
